### ISSUE TITLE
B::CC Unexpected "Use of uninitialized value" in conditional

### ISSUE THREAD

**Issue by rurban** [rurban/perl-compiler#76]
Original [issue 74](https://code.google.com/p/perl-compiler/issues/detail?id=74) created by rurban on 2011-07-06T18:56:13.000Z:

This test program prints unexpected &quot;Use of uninitialized value&quot; if compiled with B::CC.

test24.pl

``` perl
use warnings;
my $u; 
my $v;
print "t1\n";
if ($u) {
    $v = 1;
}
else {
    print "t2\n";
}
print "t3\n";
```

Expected output:
$ perl test24.pl
t1
t2
t3

With B:CC:
$ perlcc -O -o test24 test24.pl
$ ./test24
t1
t2
Use of uninitialized value in next statement at test24.c line 11.
t3

Used versions:
- perl v5.12.3
- B::C r1118

B::CC processes basic blocks in this order:
(1) print t1
(2) if clause
(3) print t3
(4) else clause with print t2
After processing (2) the if clause, it knows that $v has an integer value and
it stores a copy of this value inside a C variable.
While processing (3), the compiler still assumes the value of $v being stored in the C variable. It would generate optimized code to access $v. But this doesn't happen because $v isn't used in (3).
The problem occurs while processing (4), the else clause.
This basic block doesn't use $v, but control flows to basic block (3).
The compiler must meet the previous assumption during compilation of (3) and
therefore it loads the integer value of $v into the known C variable.
This triggers the warning message, because $v has an undefined value.
